transformation/type.go

```diff
@@ -20,12 +20,77 @@
 
 package transformation
 
+import (
+	"fmt"
+)
+
 // Type defines a transformation function.
 type Type int
 
 // Supported transformation types.
 const (
 	UnknownType Type = iota
-	PerSecond
 	Absolute
+	PerSecond
+)
+
+// IsUnaryTransform returns whether this is a unary transformation.
+func (t Type) IsUnaryTransform() bool {
+	_, exists := unaryTransforms[t]
+	return exists
+}
+
+// IsBinaryTransform returns whether this is a binary transformation.
+func (t Type) IsBinaryTransform() bool {
+	_, exists := binaryTransforms[t]
+	return exists
+}
+
+// UnaryTransform returns the unary transformation function associated with
+// the transformation type if applicable, or an error otherwise.
+func (t Type) UnaryTransform() (UnaryTransform, error) {
```

Comment:
Return `UnaryTransform, bool`?

Reply:
Feels more intuitive to return an error. `IsUnaryTransform` already returns an error.